content/blog/2022/2022-08-15-admin-operations-with-nodejs-launchdarkly.md

```diff
@@ -0,0 +1,681 @@
+```bash
+npm run rateLimiter
+```
+
+Now when an unsubscribed user hits the APIs within the defined limit he will get to see:
+
+{{% image alt="Normal API" src="images/posts/nodejs-backend-launchdarkly/Normal_API_Call.png" %}}
+
+But when the threshold limit exceeds for that user, he will see the following error:
+
+{{% image alt="Rate Limit Error" src="images/posts/nodejs-backend-launchdarkly/Rate_Limiter_Error.png" %}}
+
+### Schedule Cron Jobs Dynamically
+
+Sometimes System Admins need to schedule *cron* jobs to perform various tasks like gathering system metrics, generate important report,  clearing or archiving logs, taking backup, etc. Usually, this cron jobs are scheduled using `cronTime` which is understood by cron executors. If there is a sudden need to change the `cronTime` of a particular job, then we can define it in LaunchDarkly and use it whenever the cron runs.
+
+For this, first we will run:
+
+```bash
+npm install cron
+```
+
+Then, we will define a variation in LaunchDarkly with string parameters which will take `cronTime` as desired values.
+
+{{% image alt="Rate Limit Error" src="images/posts/nodejs-backend-launchdarkly/Cron_Variation.png" %}}
+
+Next, we will define our cron job code which will retrieve the config from Launchdarkly and schedule the cron:
+
+```javascript
+import cron from 'cron';
+import LaunchDarkly from 'launchdarkly-node-server-sdk';
+
+const CronJob = cron.CronJob;
+const CronTime = cron.CronTime;
+
+// Initiating LaunchDarkly Client
+const LD_SDK_KEY = 'sdk-d2432dc7-e56a-458b-9f93-0361af47d578';
+const userName = 'admin';
+const launchDarklyClient = LaunchDarkly.init( LD_SDK_KEY );
+
+launchDarklyClient.once('ready', async () => {    
+    const cronConfig = await launchDarklyClient.variation(
+      'cron-config',
+      {
+          key: userName
+      },
+      '*/4 * * * *' // Default fall-back variation value.
+    );
+
+    const job = new CronJob(cronConfig, function() {
+      run();
+    }, null, false)
+
+    let run = () => {
+      console.log('scheduled task called');
+    }
+
+    let scheduler = () => {
+      console.log('CRON JOB STARTED WILL RUN AS PER LAUNCHDARKLY CONFIG');
+      job.start();
+    }
+
+    let schedulerStop = () => {
+      job.stop();
+      console.log('scheduler stopped');
+    }
+
+    let schedulerStatus = () => {
+      console.log('cron status ---->>>', job.running);
+    }
+
+    let changeTime = (input) => {
+      job.setTime(new CronTime(input));
+      console.log('changed to every 1 second');
+    }
+
+    scheduler();
+    setTimeout(() => {schedulerStatus()}, 1000);
+    setTimeout(() => {schedulerStop()}, 9000);
+    setTimeout(() => {schedulerStatus()}, 10000);
+    setTimeout(() => {changeTime('* * * * * *')}, 11000);
+    setTimeout(() => {scheduler()}, 12000);
+    setTimeout(() => {schedulerStop()}, 16000);
+  }
+);
+```
+
+Next we can define the script as part of `package.json`:
+
+```json
+{
+	"scripts": {
+    	"cron": "nodemon cron_job.js"
```

Comment:
why use `nodemon` here and not `node` as in the previous examples? If there's a reason, please explain in the text, if not use `node`

Reply:
this was a typo as I was using it for my dev. But I have replaced it with node.